### PR TITLE
Fix trading taking the wrong item

### PR DESCRIPTION
**Pull request by EmosewaMC**
When trading, we need to take the item by ID and not by lot.  This caused the wrong item to be taken.

Tested that trading properly takes items by ID and not by lot.

### CODE REVIEW

**Review by aronwk-aaron — APPROVED** [review]

**Comment by Wincent01** [comment]
You say you're aborting but nothing happens to support that, no return statement etc. Maybe before doing any transaction check that it is legal.

**Review by Wincent01 — CHANGES_REQUESTED** [review]
See above comment

**Comment by EmosewaMC** [comment]
> See above comment

lol, guess I missed that.  Apologies

**Comment by EmosewaMC** [comment]
Going to reinforce coins as well.

**Comment by EmosewaMC** [comment]
Repeated the same tests, players who trade exactly enough coins or who trade exactly how many items they have can still trade, but if you somehow go over the amount in a stack, then the trade does not happen.

File: dGame/TradingManager.cpp [hunk 1]
```diff
@@ -103,6 +103,7 @@ void Trade::SetAccepted(LWOOBJID participant, bool value) {
 		}
 
 		Complete();
+		TradingManager::Instance()->CancelTrade(m_TradeId);
```
Comment (aronwk-aaron) [inline dGame/TradingManager.cpp:106]: Why is this here?

Comment (EmosewaMC) [reply dGame/TradingManager.cpp:106]: Because previously this was at the bottom of complete, however we may not always hit that.

**Review by Wincent01 — APPROVED** [review]

**Review by aronwk-aaron — APPROVED** [review]
